src/material/form-field/_mdc-text-field-structure.scss

```diff
@@ -458,15 +458,16 @@ $token-slots: m2-form-field.get-token-slots();
     }
 
     .mdc-notched-outline__notch {
+      $min-value: 'calc(100% - max(12px, #{token-utils.slot(outlined-container-shape)}) * 2)';
       flex: 0 0 auto;
       width: auto;
 
       .mdc-text-field--outlined .mdc-notched-outline & {
-        $shape-var: token-utils.slot(outlined-container-shape);
-        max-width: min(
-          var(--mat-form-field-notch-max-width, 100%),
-          calc(100% - max(12px, #{$shape-var}) * 2)
-        );
+        max-width: min(var(--mat-form-field-notch-max-width, 100%), #{$min-value});
```

Comment:
`--mat-form-field-notch-max-width` looks like a token, but I assume its just internal-only. Seems like we should have a different shape of variables that are private. Like `--_mat_private-` or something

Reply:
Yeah not really a token, just an implementation detail.